src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -361,3 +429,60 @@ def generate_grant_sql_ddl_statements(
         grant_sql_statements.append(grant_sql_statement)
 
     return "\n".join(grant_sql_statements)
+
+
+def generate_new_sql_file_name(
```

Comment:
nit: move this to the class, and access deploy and generated roots directly from `self`?

Reply:
Speculative complexity. We probably wouldn't reuse the helper from there, we might want to introduce a base class for generation-based logic, etc. IMHO write the simplest code, most readable code there is, and refactor aggressively.